host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -1139,14 +1140,76 @@ static void update_annotation_data(RASPIVID_STATE *state)
    if (state->camera_parameters.enable_annotate & ANNOTATE_APP_TEXT)
    {
       char *text;
-      const char *refresh = raspicli_unmap_xref(state->intra_refresh_type, intra_refresh_map, intra_refresh_map_size);
 
-      asprintf(&text,  "%dk,%df,%s,%d,%s,%s",
-               state->bitrate / 1000,  state->framerate,
-               refresh ? refresh : "(none)",
-               state->intraperiod,
-               raspicli_unmap_xref(state->profile, profile_map, profile_map_size),
-               raspicli_unmap_xref(state->level, level_map, level_map_size));
+      if (state->common_settings.gps)
+      {
+         char lat[24] = {"n/a"};
+         char lon[24] = {"n/a"};
+         char alt[24] = {"Altitude n/a"};
+         char speed[24] = {"Speed n/a"};
+         char track[24] = {"Track n/a"};
+         char datetime[32] = {"Time n/a"};
+
+         struct gps_data_t *gpsdata = raspi_gps_lock();
+
+         if (gpsdata->set & TIME_SET)
+         {
+            time_t rawtime;
+            struct tm *timeinfo;
+            rawtime = gpsdata->fix.time;
+            timeinfo = localtime(&rawtime);
+            strftime(datetime, sizeof(datetime), "%Y:%m:%d %H:%M:%S", timeinfo);
+         }
+
+
+         if (gpsdata->online && gpsdata->fix.mode >= MODE_2D)
+         {
+            if (gpsdata->set & LATLON_SET)
+            {
+               if (isnan(gpsdata->fix.latitude) == 0)
+               {
+                  snprintf(lat, sizeof(lat), "%.6lf", gpsdata->fix.latitude);
+               }
+               if (isnan(gpsdata->fix.longitude) == 0)
+               {
+                  snprintf(lon, sizeof(lon), "%.6lf", gpsdata->fix.longitude);
+               }
+            }
```

Comment:
If I were writing this it would be !isnan and no braces, but as I've been reminded, this isn't the kernel (which means the braces are in the right place).

Reply:
Have removed some but not all, for reasons of readability rather than consistency.